src/GAP/utils.jl

```diff
@@ -0,0 +1,11 @@
+function _write_gap_file(relpath::String, content::String)
+  gaproot = GAP.Setup.gaproot()
+  path = joinpath(gaproot, relpath)
+  mkpath(dirname(path))
+  # write to a temporary file and move later to avoid race conditions
+  tmpfile = tempname(; cleanup=false)
+  open(tmpfile, "w") do file
+    write(file, content)
+  end
+  Base.Filesystem.mv(tmpfile, path; force=true)
```

Comment:
```suggestion
  tmpfile = tempname(dirname(path); cleanup=false)
  open(tmpfile, "w") do file
    write(file, content)
  end
  Base.Filesystem.rename(tmpfile, path)
```
To make this race-condition free you need to put the temporary file in the same directory as the target file (the same filesystem would suffice but directory is easier) and then use the undocumented `rename` function.